src/overlays/actors/ovl_Dm_Al/z_dm_al.c

```diff
@@ -30,24 +27,133 @@ const ActorInit Dm_Al_InitVars = {
+        }
+    } else if (this->unk45C != 0) {
+        this->unk45C = 0;
+        func_80C1BD90(this, this->animationIndex2);
+    }
+}
+
+void DmAl_Init(Actor* thisx, PlayState* play) {
+    DmAl* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gMadameAromaSkel, NULL, this->jointTable, this->morphTable,
+                       OBJECT_AL_LIMB_MAX);
+    this->animationIndex = -1;
+    func_80C1BD90(this, NULL);
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = func_80C1BDD8;
+}
+
+void DmAl_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void DmAl_Update(Actor* thisx, PlayState* play) {
+    DmAl* this = THIS;
+
+    this->actionFunc(this, play);
+    SkelAnime_Update(&this->skelAnime);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 30.0f, 12.0f, 0.0f, 4);
+}
+
+s32 func_80C1C028(PlayState* play, s32 limbIndex, Gfx** gfx, Vec3f* rot, Vec3s* pos, Actor* thisx) {
+    switch (limbIndex) {
+        case MADAME_AROMA_SHAWL_MIDDLE_LIMB:
+        case MADAME_AROMA_SHAWL_UPPER_LIMB:
+        case MADAME_AROMA_SHAWL_LEFT_LOWER_MIDDLE_LIMB:
+        case MADAME_AROMA_SHAWL_LEFT_LOWER_LIMB:
+        case MADAME_AROMA_SHAWL_RIGHT_LOWER_MIDDLE_LIMB:
+        case MADAME_AROMA_SHAWL_RIGHT_LOWER_LIMB:
+            *gfx = NULL;
+            break;
+    }
+    return false;
+}
+
+void func_80C1C064(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    DmAl* this = THIS;
+
+    switch (limbIndex) {
+        case MADAME_AROMA_SHAWL_MIDDLE_LIMB:
+            Matrix_Get(&this->unkMtx[0]);
+            break;
+        case MADAME_AROMA_SHAWL_UPPER_LIMB:
+            Matrix_Get(&this->unkMtx[1]);
+            break;
+        case MADAME_AROMA_SHAWL_LEFT_LOWER_MIDDLE_LIMB:
+            Matrix_Get(&this->unkMtx[2]);
+            break;
+        case MADAME_AROMA_SHAWL_LEFT_LOWER_LIMB:
+            Matrix_Get(&this->unkMtx[3]);
+            break;
+        case MADAME_AROMA_SHAWL_RIGHT_LOWER_MIDDLE_LIMB:
+            Matrix_Get(&this->unkMtx[4]);
+            break;
+        case MADAME_AROMA_SHAWL_RIGHT_LOWER_LIMB:
+            Matrix_Get(&this->unkMtx[5]);
+            break;
+        default:
+            break;
+    }
+}
+
+void func_80C1C11C(PlayState* play, s32 arg0, Actor* thisx) {
+}
+
+s32 sDlists[6] = { object_al_DL_006598, object_al_DL_005920, object_al_DL_005878,
+                   object_al_DL_0057D0, object_al_DL_005728, object_al_DL_005680 };
+s32 D_80C1C2AC = 0;
+
+void DmAl_Draw(Actor* thisx, PlayState* play) {
+    u32 i;
+    DmAl* this = THIS;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    func_8012C28C(play->state.gfxCtx);
+    SkelAnime_DrawTransformFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable,
+                                   this->skelAnime.dListCount, func_80C1C028, func_80C1C064, func_80C1C11C,
```

Comment:
```suggestion
    SkelAnime_DrawTransformFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable,
                                   this->skelAnime.dListCount, DmAl_OverrideLimbDraw, DmAl_PostLimbDraw, DmAl_TransformLimbDraw,
```
Can name these based on the function call.